### ISSUE TITLE
feat: junior curriculum progress tracking

### ISSUE THREAD

**Issue by matt-d-webb** [chess-centre/platform#447]
## User Story
**Given** a parent usually creates a junior membership
**When** they log into the Chess Centre site
**Then** they see the current learning progress of their child(ren)

### Technical challenge

This requires logged attendence and records kept of the lessons taken. A data structure for this might look as follows:

```
{
   "id": uuid,
   "memberId": string<member>
   "lessonId": string<lesson>
   "completedDate": datetime
}
```
Here a lesson object might look as follows:
```
{
   "id": uuid,
   "name": string, // Rook & Pawn Endings
   "description": string, 
   "S3Object": ref,
   "tags": array
}
```
It is key to distinguish the lessons from when these have been recorded as "complete". Providing an ability to extract any lesson details to either catch up or as a refresher.

**Comment by matt-d-webb** [comment]
@CatVarley good to get your input on lesson plan materials.

**Comment by CatVarley** [comment]
We spoke about a simple PDF format.
Another idea that could scale up well or be reused and added to easily is Google Slides. Opportunity to follow a similar format each time if we set up a base template (branded & colour matched). 
Good spot to deposit links and other sources. Can naturally build a flow to a lesson, cue branch out sessions. Also flexible enough to skip through bits as deemed necessary in lesson

**Comment by CatVarley** [comment]
<img width="726" alt="Screen Shot 2021-12-22 at 15 26 45" src="https://user-images.githubusercontent.com/16880382/147115845-f56a4b0d-8401-438a-853f-ff80bc9c2bd1.png">
<img width="1210" alt="Screen Shot 2021-12-22 at 14 22 17" src="https://user-images.githubusercontent.com/16880382/147115864-40263059-d408-4070-a003-6886c86b026b.png">
<img width="1199" alt="Screen Shot 2021-12-22 at 14 21 33" src="https://user-images.githubusercontent.com/16880382/147115871-3d4c6fd4-873b-4838-b57
<img width="1216" alt="Screen Shot 2021-12-22 at 14 21 55" src="https://user-images.githubusercontent.com/16880382/147115893-0e40e251-53ab-4e0b-9e13-4463db1fa0b8.png">
3-e7d73e160964.png">

These are simple phonics slides - replace with the key element of the game / piece you're working on. 
Start with objective of session
Opportunity to recap basics / previous learning (judge within a session - can be skipped)
Add problems - gathered from other existing sources - discuss in group
Branch out - practice from given position, timed finish, game etc
Regroup - Share positions found in game. Visit more problems 
Assessment - Self score - revisiting objective - Are you now confident in X? How will this impact your play?

**Comment by CatVarley** [comment]
To keep a track of who has attended each session maybe we could merge ideas from different apps we use in school (I'll add some screen shots but I need to figure out how to mask children's names for some I'll add later). 
If it's possible to build tabs for each child - so you select each one to say they are there. You could then select the lesson / or add objective to the whole group (link to lesson slides?) 
We also use a nice messaging service to keep in touch with parents. This could be used to reward hard work in chess lessons and also as direct messaging....

**Comment by CatVarley** [comment]
<img width="1257" alt="Screen Shot 2021-12-22 at 14 01 28" src="https://user-images.githubusercontent.com/16880382/147118424-2b21bc0d-17d5-4454-840a-c5cc57b213d5.png">
![Uploading Screen Shot 2021-12-22 at 14.01.46.png…]()

**Comment by CatVarley** [comment]
<img width="1257" alt="Screen Shot 2021-12-22 at 14 01 46" src="https://user-images.githubusercontent.com/16880382/147118479-aa355f3e-51b6-4136-b1a2-8eac380e6523.png">

**Comment by CatVarley** [comment]
For assessment logging, we use a rather onerous drop-down graph. However, the data it then produces is useful so it's kind of worth it in the end. Whether we'd need this much detail for TCC could be discusse
<img width="881" alt="Screen Shot 2021-12-22 at 15 47 59" src="https://user-images.githubusercontent.com/16880382/147119091-4a2c25bf-c7b6-4f4c-b90f-658a1fb96bbc.png">
d

**Comment by CatVarley** [comment]
Some mock session slides...
<img width="907" alt="Screen Shot 2021-12-22 at 17 01 57" src="https://user-images.githubusercontent.com/16880382/147128839-30318de6-3b5c-4060-9f5a-750552dfbeb9.png">
<img width="904" alt="Screen Shot 2021-12-22 at 17 02 04" src="https://user-images.githubusercontent.com/16880382/147128858-402854a7-4dad-42e0-b916-f6cb3225fa7a.png">

**Comment by matt-d-webb** [comment]
Love these! :-)

**Comment by CatVarley** [comment]
In terms of longer-term planning, I've edited this table so it would fit just one topic. The term timescales I've used are based on the Bradford Catholic schools. We could look at the next academic year and the local Ilkley school's calendars to make it fit. May be worth planning lessons in term time and fitting larger scale junior tournaments in the half term - or perhaps that last session of each term if many children are away in the holidays?
<img width="767" alt="Screen Shot 2021-12-23 at 11 09 25" src="https://user-images.githubusercontent.com/16880382/147232533-5deb7fcb-16d0-4f65-ba7e-57c58a4d58df.png">

**Comment by CatVarley** [comment]
In this table, you'd put a very general overview of what you'll be covering. Details come in the weekly planning. This would also be a working document - for example, you'd planned to spend two weeks on a topic but they are hot on it straight away you can change that second week and vice versa, if they do not understand a concept, you can extend it. Worth editing on the plan for the following years planning.

**Comment by matt-d-webb** [comment]
@CatVarley - good to have knowledge of the term time table. Many of the juniors do go on holiday between terms so planning things around that is useful.

Given we wont always stricktly take each child through lesson 1 to N, and we may also have N children doing N lessons in a single coaching leason.

i.e we have 3 new joiners mixed in with 3 experienced players already at level say, 10. We want to be able to select from a range of "ability level basics" which we can cover.

**Comment by CatVarley** [comment]
True! Definitely need to consider catering for all abilities within a session. 
Worth noting, if parents see a progression - you could make an overview plan available to them at the beginning of each term (we do this in school), they may place more value in bringing them every week instead of seeing it as a drop in.

**Comment by CatVarley** [comment]
Maybe the long term plan could be used for the more experienced level players - the deeper thinking. We could build a beginners/intro to chess club program to bolt on the side. So those that join can start from there each time (or where within it they need to be) then be immersed in the experienced planning once they are on track?

**Comment by matt-d-webb** [comment]
Refencing discussion topic link: https://github.com/chess-centre/platform/discussions/455

**Comment by matt-d-webb** [comment]
Quick overview of potential dashboard:
![image](https://user-images.githubusercontent.com/36933715/164710155-40088ffd-cc02-42c8-9bc8-42c2bf88ad3f.png)
